### ISSUE TITLE
Limited marker functionality - no easy solution for popovers

### ISSUE THREAD

**Issue by oatssss** [ajaxorg/ace#3135]
Hi, I'm trying to extend the functionality of the editor's markers. Specifically, I'm trying to use react to achieve bootstrap-like popups when hovering over text markers. It seems there's no easy way to do this presently, but with a minor change to the Marker render process, this can be done in a fairly straightforward way.

The line in question is this one:
https://github.com/ajaxorg/ace/blob/ca0953efebb881f049cbf29e8a328aa5a33ad410/lib/ace/layer/marker.js#L96
and a PR with the full 2-line change required to make this work:
https://github.com/ajaxorg/ace/pull/3143/files

Here's a codepen showing off the kinds of things that can be done once this fix is in place (the codepen uses a custom build of Ace with insertAdjacentHTML).
https://codepen.io/oatssss/pen/oYxJQV

In this case, the change must be made so that modifications to the marker layer from within a dynamic marker's update, aren't eventually erased by the layer's assignment to innerHTML.

So how is this beneficial to Ace as a whole?
I find it's much more useful to developers integrating ace in their projects to be able to modify the editor's marker DOM in a manner of their choosing (react included), than being forced to use the current method of passing raw strings of html to a string builder.

Additionally, I would love to hear any other suggestions on how to achieve marker popups without modifying Ace's internals.

**Comment by AlonBe** [comment]
I'm encountering the same issue.
@oatssss's PR is well explained

Any news regarding this issue?

**Comment by elk941** [comment]
@oatssss have you looked at [this](https://codepen.io/oatssss/pen/oYxJQV/) ?

**Comment by akoreman** [comment]
Closed as duplicate of #2720.
